client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -18,42 +19,42 @@ import Dialog from 'components/dialog';
 import CancelPurchaseForm from 'components/marketing-survey/cancel-purchase-form';
 import { getName, getSubscriptionEndDate, isOneTimePurchase, isRefundable, isSubscription } from 'lib/purchases';
 import { enrichedSurveyData } from '../utils';
-import { isDomainRegistration, isTheme, isGoogleApps, isJetpackPlan } from 'lib/products-values';
+import { isDomainRegistration, isJetpackPlan } from 'lib/products-values';
 import notices from 'notices';
 import paths from 'me/purchases/paths';
 import { refreshSitePlans } from 'state/sites/plans/actions';
 import FormSectionHeading from 'components/forms/form-section-heading';
 import { recordTracksEvent } from 'state/analytics/actions';
+import { cancellationEffectDetail, cancellationEffectHeadline } from './cancellation-effect';
 
-const CancelPurchaseButton = React.createClass( {
-	propTypes: {
-		purchase: React.PropTypes.object.isRequired,
-		selectedSite: React.PropTypes.object.isRequired
-	},
+class CancelPurchaseButton extends Component {
+	static propTypes = {
+		purchase: PropTypes.object.isRequired,
+		selectedSite: PropTypes.object.isRequired
+	}
 
-	getInitialState() {
-		return {
-			disabled: false,
-			showDialog: false,
-			isRemoving: false,
-			surveyStep: 1,
-			survey: {
-				questionOneRadio: null,
-				questionTwoRadio: null
-			}
-		};
-	},
+	state = {
+		disabled: false,
+		showDialog: false,
+		isRemoving: false,
+		surveyStep: 1,
+		survey: {
+			questionOneRadio: null,
+			questionTwoRadio: null
+		}
+	}
 
-	recordEvent( name, properties = {} ) {
-		const product_slug = get( this.props, 'purchase.productSlug' );
-		const refund = true;
+	recordEvent = ( name, properties = {} ) => {
```

Comment:
Optional change, but this can use the normal class function syntax since it doesn't seem to be called anywhere without `this` bound.